CMakeLists.txt

```diff
@@ -58,7 +58,13 @@ function(cctz_target_set_cxx_standard target)
   endif()
 endfunction()
 
-add_library(cctz
+set(_HDRS
+  include/cctz/time_zone.h
+  include/cctz/civil_time_detail.h
+  include/cctz/zone_info_source.h
+  include/cctz/civil_time.h
+  )
+set(_SRCS
```

Comment:
Note: in my initial CMake patch, I purposefully did not make such variables, because I consider it not "Modern CMake", see Stephen Kelly talk:
- Embracing Modern CMake: https://steveire.wordpress.com/2017/11/05/embracing-modern-cmake/
- Slides 49-52, "Reliance on variables": http://www.steveire.com/embracing-modern-cmake.pdf

Reply:
agree with directly adding sources in `add_library()` command directly or using `target_sources()`. I was lazy on this (often use GLOB recurse and no incremental build but containerized build from scratch).
I would like to keep headers also (didn't see any reference about it in Modern CMake) so IDE display them in there solution (while it is not strictly needed for the compilation)

Reply:
1) [PUBLIC_HEADER](https://cmake.org/cmake/help/latest/prop_tgt/PUBLIC_HEADER.html) is also working outside macOS, look at the last sentence:

> On non-Apple platforms these headers may be installed using the PUBLIC_HEADER option to the install(TARGETS) command.

Thus it avoid the `install(FILES` and keep you in the Modern CMake way (IMHO).
This is also mentioned is the recent CMake documentation for [`install(TARGETS ...)`](https://cmake.org/cmake/help/latest/command/install.html#targets)

> PUBLIC_HEADER
Any PUBLIC_HEADER files associated with a library are installed in the destination specified by the PUBLIC_HEADER argument on non-Apple platforms.

I still have concern if we'll get a 'target_headers()' or similar in CMake later
note: no answer from kitware https://cmake.org/pipermail/cmake/2017-March/065126.html

1bis) if you set headers as `PUBLIC` target sources they'll also appear on any target linking against this target (e.g. executable target B will show library A headers in its source list)
I'll try to take screenshot on this...

2) **QtCreator** and **VisualStudio 2017** both support as built-in cmake project and will display headers if they are listed in the target [SOURCES](https://cmake.org/cmake/help/latest/prop_tgt/SOURCES.html)
note: need to try/confirm Eclipse and CodeBlocks...

2bis) This is not working for INTERFACE currently (at least on qtcreator)
https://bugreports.qt.io/browse/QTCREATORBUG-15677
https://gitlab.kitware.com/cmake/cmake/merge_requests/2282

Reply:
So, do IDEs make use of `PUBLIC_HEADER`, or only `SOURCES`? Because I'm not advocating to remove the headers from the CMakeLists.txt, I'm just wondering what is the benefits of specifying them as `PUBLIC_HEADER`, over just listing them in the `add_library`.

In my initial patch I did not use `install(FILES` , but `install(DIRECTORY`, since all public headers are in the same directory.

I could see the benefit of `PUBLIC_HEADER` approach when using a "merged header placement" source layout:
- https://api.csswg.org/bikeshed/?force=1&url=https://raw.githubusercontent.com/vector-of-bool/pitchfork/spec/data/spec.bs#src.header-placement.merged

Anyway, I think it would look slightly cleaner, but equally correct so I don't really mind.

Reply:
You can take a look at https://github.com/Mizux/plop/blob/master/README.md
sources PUBLIC need generator like include (otherwise you'll get cmake error)
ps: open image in new tab

Reply:
Thanks for the example.
I'm not sure `target_sources`'s `PUBLIC` is meant for public headers, I would be interested to know, but it feels off topic for this PR.

Reply:
Agree can be done later.
ps: if you get info on this topic I will be happy if you can share it with me.
pps: I also done few test in plop project to understand the effect of install rule on public syntax
TLDR:
```cmake
if(ENABLE_INSTALL)
  target_sources(B
    PRIVATE
    src/B.cc
    PUBLIC
    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/b/B.h>
    $<INSTALL_INTERFACE:include/b/B.h>
    )
else()
  target_sources(B
    PRIVATE
    src/B.cc
    PUBLIC
    ${CMAKE_CURRENT_SOURCE_DIR}/include/b/B.h
    )
endif()
```
i.e. like target_include_directory() you need to use `(BUILD|INSTALL)_INTERFACE:` generator once you start to have install rules...